aea/manager/manager.py

```diff
@@ -48,6 +50,38 @@ def __init__(self, msg: str, source_exception: Exception):
         self.source_exception = source_exception
 
 
+class ProjectPackageConsistencyCheckError(ValueError):
+    """Check consistency of package versions against already added project."""
+
+    def __init__(
+        self,
+        agent_project_id: PublicId,
+        conflicting_packages: List[Tuple[PackageIdPrefix, str, str, Set[PublicId]]],
+    ):
+        """
+        Initialize the exception.
+
+        :param agent_project_id: the agent project id whose addition has failed.
+        :param conflicting_packages: the conflicting packages.
+        """
+        self.agent_project_id = agent_project_id
+        self.conflicting_packages = conflicting_packages
+        super().__init__(self._build_error_message())
+
+    def _build_error_message(self) -> str:
```

Comment:
for an example of the generated message, see below:
https://github.com/fetchai/agents-aea/pull/2550/files#diff-5a86c9151c53eb9b7fde346e231dc88a7b18f51ff369081b9116ef1c02423bddR604